include/cudaq/Optimizer/Dialect/Quake/QuakeOps.td

```diff
@@ -786,6 +786,23 @@ class TwoTargetOp<string mnemonic, list<Trait> traits = []> :
 // Quantum operators (gates)
 //===----------------------------------------------------------------------===//
 
+def ExpPauliOp : QuakeOp<"exp_pauli", []> {
+  let summary = "General Pauli tensor product rotation";
+  let description = [{
+    This operation affects a general Pauli tensor product rotation on 
+    the input qubits. The number of Pauli characters in the input Pauli word 
+    string must equal the number of qubits in the veq. Mathematically, this operation 
+    applies exp(i theta P) where P is a general Pauli tensor product. 
+  }];
+  
+  let arguments = (ins AnyFloat:$parameter, VeqType:$qubits, cc_PointerType:$pauli);
```

Comment:
Noting that this should be extended to allow wires, etc.

Reply:
I will create a tracker for it. I will need your help on it.